dlc-manager/src/manager.rs

```diff
@@ -1986,14 +2001,50 @@ where
         Ok(())
     }
 
-    fn on_reject(&mut self, reject: &Reject, counter_party: &PublicKey) -> Result<(), Error> {
-        let mut signed_channel =
-            get_channel_in_state!(self, &reject.channel_id, Signed, Some(*counter_party))?;
+    fn on_reject(&self, reject: &Reject, counter_party: &PublicKey) -> Result<(), Error> {
+        let channel = self.store.get_channel(&reject.channel_id)?;
+
+        if let Some(channel) = channel {
+            if channel.get_counter_party_id() != *counter_party {
+                return Err(Error::InvalidParameters(format!(
+                    "Peer {:02x?} is not involved with {} {:02x?}.",
+                    counter_party,
+                    stringify!(Channel),
+                    channel.get_id()
+                )));
+            }
+            match channel {
+                Channel::Offered(offered_channel) => {
+                    let offered_contract = get_contract_in_state!(self, &offered_channel.offered_contract_id, Offered, None as Option<PublicKey>)?;
+                    let utxos = offered_contract.funding_inputs_info.iter().map(|funding_input_info| {
+                        let txid = Transaction::consensus_decode(&mut funding_input_info.funding_input.prev_tx.as_slice())
+                            .expect("Transaction Decode Error")
+                            .txid();
+                        let vout = funding_input_info.funding_input.prev_tx_vout;
+                        OutPoint{txid, vout}
+                    }).collect::<Vec<_>>();
+
+                    self.wallet.unreserve_utxos(&utxos)?;
+
+                    // remove rejected channel, since nothing has been confirmed on chain yet.
+                    self.store.upsert_channel(Channel::Cancelled(offered_channel), Some(Contract::Rejected(offered_contract)))?;
+                },
+                Channel::Signed(mut signed_channel) => {
+                    crate::channel_updater::on_reject(&mut signed_channel)?;
 
-        crate::channel_updater::on_reject(&mut signed_channel)?;
+                    self.store
+                        .upsert_channel(Channel::Signed(signed_channel), None)?;
```

Comment:
I think it was like that before, but I now realized that in the case of rejecting in `RenewOffer` the offered contract state should also be rejected.

Reply:
300cbf1dd884f0fa8dcbfb6356604699880f06e0